examples/Migration/CampaignManagement/CreateCompleteCampaignBothApisPhase4.php

```diff
@@ -19,10 +19,448 @@
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        return $response->getIterator()->current()->getCampaign();
+    }
+
+    /**
+     * Creates an ad group.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param Campaign $campaign the campaign
+     * @return AdGroup the newly created ad group
+     */
+    private static function createAdGroup(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        Campaign $campaign
+    ) {
+        // Constructs an ad group and sets an optional CPC value.
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars Cruises #' . uniqid()]),
+            'campaign' => $campaign->getResourceName(),
+            'status' => AdGroupStatus::ENABLED,
+            'type' => AdGroupType::SEARCH_STANDARD,
+            'cpc_bid_micros' => new Int64Value(['value' => 10000000])
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Issues a mutate request to add the ad groups.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $adGroupResponse = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        $adGroupResourceName = $adGroupResponse->getResults()[0]->getResourceName();
+        $newAdGroup = self::getAdGroup($googleAdsClient, $customerId, $adGroupResourceName);
+
+        printf("Added ad group named '%s'.%s", $newAdGroup->getName()->getValue(), PHP_EOL);
+
+        return $newAdGroup;
+    }
+
+    /**
+     * Gets an ad group.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param string $resourceName the resource name of the ad group to retrieve
+     * @return AdGroup the ad group
+     */
+    private static function getAdGroup(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        string $adGroupResourceName
+    ) {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $query = "SELECT ad_group.id, ad_group.name, ad_group.resource_name " .
+                 "FROM ad_group " .
+                 "WHERE ad_group.resource_name = '$adGroupResourceName'";
+
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        return $response->getIterator()->current()->getAdGroup();
+    }
+
+    /**
+     * Creates text ads.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param AdGroup $adGroup the ad group
+     * @return AdGroupAd[] an array of text ads
```

Comment:
Also update this PHPDoc. You don't need to return this anymore. Please update all files.

Reply:
Done.